app/main.py

```diff
@@ -1,3 +1,23 @@
-def cache(func):
-    # Write your code here
-    pass
+from functools import wraps
+from typing import Callable, Any
+
+
+def cache(func: Callable) -> Callable:
+    caching_result = []
+
+    @wraps(func)
+    def wrapper(*args) -> Any:
+        for item in caching_result:
+            if args in item:
```

Comment:
You could limit this expression to `if args in caching_result` in case you would use `dict`.